codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -151,20 +150,16 @@ public boolean goAwayReceived() {
     }
 
     @Override
-    public void goAwayReceived(final int lastKnownStream, long errorCode, ByteBuf debugData) {
+    public void goAwayReceived(final int lastKnownStream, long errorCode, ByteBuf debugData) throws Http2Exception {
         localEndpoint.lastStreamKnownByPeer(lastKnownStream);
-        for (int i = 0; i < listeners.size(); ++i) {
-            try {
+        try {
+            for (int i = 0; i < listeners.size(); ++i) {
                 listeners.get(i).onGoAwayReceived(lastKnownStream, errorCode, debugData);
-            } catch (RuntimeException e) {
-                logger.error("Caught RuntimeException from listener onGoAwayReceived.", e);
             }
-        }
 
-        try {
             forEachActiveStream(new Http2StreamVisitor() {
                 @Override
-                public boolean visit(Http2Stream stream) {
+                public boolean visit(Http2Stream stream) throws Http2Exception {
```

Comment:
`stream.close();` can now propagate exceptions because it can call a listener method ... which can throw an exception.